spec/neo4j/core/cypher_session/adaptors/http_spec.rb

```diff
@@ -22,6 +23,34 @@
       expect { adaptor_class.new('bolt://localhost:7474').connect }.to raise_error ArgumentError, /Invalid URL/
       expect { adaptor_class.new('foo://localhost:7474').connect }.to raise_error ArgumentError, /Invalid URL/
     end
+
+    describe 'the faraday_options param' do
+      describe 'the adapter option' do
+        it 'uses net_http_persistent by default' do
+          expect_any_instance_of(Faraday::Connection).to receive(:adapter).with(:net_http_persistent)
+          adaptor_class.new(url).connect
+        end
+
+        it 'will pass through a symbol key' do
+          expect_any_instance_of(Faraday::Connection).to receive(:adapter).with(:something)
+          adaptor_class.new(url, faraday_options: {adapter: :something}).connect
+        end
+
+        it 'will pass through a string key' do
+          expect_any_instance_of(Faraday::Connection).to receive(:adapter).with(:something)
+          adaptor_class.new(url, 'faraday_options' => {'adapter' => :something}).connect
```

Comment:
Yes, maybe that's a bad test but I was only testing that a parameter was passed through, not that the parameter was valid.  I guess I can send in a valid parameter here (assuming I understand correctly).